timeseries/src/autogluon/timeseries/models/chronos/pipeline/utils.py

```diff
@@ -1,14 +1,210 @@
+import logging
 import os
 import re
 import time
 from pathlib import Path
-from typing import Callable, List, Optional
+from typing import TYPE_CHECKING, Callable, Iterable, Iterator, List, Literal, Optional
 
 import numpy as np
 import torch
+from gluonts.dataset.field_names import FieldName
+from gluonts.transform import ExpectedNumInstanceSampler, InstanceSplitter, ValidationSplitSampler
+from torch.utils.data import IterableDataset
+from transformers import TrainerCallback
 
 from autogluon.core.utils.exceptions import TimeLimitExceeded
 from autogluon.timeseries.dataset.ts_dataframe import TimeSeriesDataFrame
+from autogluon.timeseries.models.gluonts.abstract_gluonts import SimpleGluonTSDataset
+
+if TYPE_CHECKING:
```

Comment:
very cool. although if I was to nit, the fact that this circular dependency is here hints that ChronosFineTuningDataset should not depend on the Tokenizer, but rather the pipeline object should know about how it will do tokenization. we can drop a TODO for later.

Reply:
Left a TODO.